libs/core/langchain_core/utils/pydantic.py

```diff
@@ -183,6 +122,9 @@ def pre_init(func: Callable) -> Any:
     with warnings.catch_warnings():
         warnings.filterwarnings(action="ignore", category=PydanticDeprecationWarning)
 
+        # Ideally we would use @model_validator(mode="before") but this would change the
+        # order of the validators. See https://github.com/pydantic/pydantic/discussions/7434.
+        # So we keep root_validator for backward compatibility.
```

Comment:
One possibility would be to deprecate `pre_init` and create a new decorator, for instance `before_init`, that uses `model_dump`.
WDYT ?